src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -196,6 +199,30 @@ internal TracerProviderSdk AddProcessor(BaseProcessor<Activity> processor)
             return this;
         }
 
+        /// <summary>
+        /// Called by <c>Shutdown</c>. This function should block the current
+        /// thread until shutdown completed or timed out.
+        /// </summary>
+        /// <param name="timeoutMilliseconds">
+        /// The number of milliseconds to wait, or <c>Timeout.Infinite</c> to
+        /// wait indefinitely.
+        /// </param>
+        /// <returns>
+        /// Returns <c>true</c> when shutdown succeeded; otherwise, <c>false</c>.
+        /// </returns>
+        /// <remarks>
+        /// This function is called synchronously on the thread which made the
+        /// first call to <c>Shutdown</c>. This function should not throw
+        /// exceptions.
+        /// </remarks>
+        internal bool OnShutdown(int timeoutMilliseconds)
+        {
+            bool? result;
```

Comment:
why not dispose the instrumentations?

Reply:
The spec just says that any calls to TracerPorvider after Shutdown should result in a no-op. It also explicitly says to call shutdown of each of the processors. We can achieve this by only disposing the listener and calling the CompositeProcessor Shutdown. I didn't dispose instrumentation to keep the Shutdown functionality as minimum as possible and in accordance with the spec.

We could Dispose the instrumentations too and then TracerProviderSdk Shutdown will be functionally pretty close to TracerProviderSdk Dispose with only Sampler and the base TracerProvider not getting disposed.

Reply:
Spec just says *at least* processors must be shutdown.
i think shutdown should do same as dispose.